### ISSUE TITLE
Document the test-running architecture in more detail, and perhaps put the tests on GitHub?

### ISSUE THREAD

**Issue by ExplodingCabbage** [Fyrd/caniuse#3196]
Although I can see what feature tests you guys are using by viewing the source at http://tests.caniuse.com/, it's not a wholly obvious or convenient way to view the tests, and the only contributing method described at http://tests.caniuse.com/submit.html is to email in tests. Wouldn't it be better for tests to be part of this repository, and for new tests to be submitted by PR? This would make the process more visible and permit more discussion.

It would also be useful for contributors to know what system you have (if any) for running the tests on old browsers. The "about" at http://tests.caniuse.com/ suggests that the test suite is for testing *upcoming* browsers, which *implies* to me that you don't have a mechanism for running all tests across old browsers (using a tool like Sauce Labs), and that as such adding a test for a feature won't be sufficient to automatically generate a historical support table... but I'm not sure if I'm inferring too much. It would be nice to have this point spelt out explicitly.

**Comment by lpd-au** [comment]
Is the email listed there actually correctly configured? Never got a response when I submitted a test. I expect there'd be alot of interest if providing merge-approved automatic-style tests meant triggering a browserstack job and immediate inclusion of new features if the data matched.

**Comment by ExplodingCabbage** [comment]
Yeah, I share your intuition about this, @lpd-au. The low interest in contributing to the tests that @Fyrd cites is probably a *direct effect* of them not being on GitHub; I'm pretty sure that people are generally much more willing to make public contributions to projects than they are to send private emails to individual project maintainers.

**Comment by danielcompton** [comment]
I agree, It would be nice to be able to add feature tests when adding a new feature. it would be much easier to determine them that way rather than having to search through different browsers support documents.

**Comment by Schweinepriester** [comment]
Really wish we would be able to add and edit tests :)
